src/main/java/org/apache/sysds/lops/MatMultCP.java

```diff
@@ -73,15 +73,15 @@ public String toString() {
 	@Override
 	public String getInstructions(String input1, String input2, String output) {
 		if(!useTranspose) {
-			return InstructionUtils.concatOperands(getExecType().name(),
+			InstructionUtils.concatBaseOperands(getExecType().name(),
```

Comment:
I don't like this change because it hides where the variable is.
also i'm not sure if it works, since strings behave differently in java than other variables (immutable),

Reply:
I have done this in other methods and I have not yet experienced any problems with it. The issue with the hidden variable is that it does not make any sense to return it from the concatBaseOperands method because of the internals of InstructionUtils. I think it is working with a thread-local variable and this means that for subsequent calls from this getInstructions method, it will always access the same variable.